core/services/ocr2/delegate.go

```diff
@@ -413,78 +411,40 @@ func (d Delegate) ServicesForSpec(jobSpec job.Job) ([]job.ServiceCtx, error) {
-		}
-
 		jobSpec.KeeperSpec = &job.KeeperSpec{
 			ID:              spec.ID,
 			ContractAddress: ethkey.MustEIP55Address(spec.ContractID),
 			FromAddress:     ethkey.MustEIP55Address(spec.TransmitterID.String),
 		}
 
-		registrySynchronizer := keeper.NewRegistrySynchronizer(keeper.RegistrySynchronizerOptions{
-			Job:                      jobSpec,
-			RegistryWrapper:          *registryWrapper,
-			ORM:                      orm,
-			JRM:                      d.jobORM,
-			LogBroadcaster:           chain.LogBroadcaster(),
-			SyncInterval:             chain.Config().KeeperRegistrySyncInterval(),
-			MinIncomingConfirmations: minIncomingConfirmations,
-			Logger:                   lggr,
-			SyncUpkeepQueueSize:      chain.Config().KeeperRegistrySyncUpkeepQueueSize(),
-			NewTurnEnabled:           chain.Config().KeeperTurnFlagEnabled(),
-		})
-
-		oracle, err := libocr2.NewOracle(libocr2.OracleArgs{
+		rgstry, err := ocr2keeperschain.NewEVMRegistryV1_2(ethkey.MustEIP55Address(spec.ContractID).Address(), chain.Client())
+		if err != nil {
+			return nil, errors.Wrap(err, "could not create new evm registry")
+		}
+		conf := ocr2keepers.DelegateConfig{
 			BinaryNetworkEndpointFactory: peerWrapper.Peer2,
 			V2Bootstrappers:              bootstrapPeers,
 			ContractTransmitter:          keeperProvider.ContractTransmitter(),
 			ContractConfigTracker:        keeperProvider.ContractConfigTracker(),
-			Database:                     ocrDB,
+			KeepersDatabase:              ocrDB,
 			LocalConfig:                  lc,
 			Logger:                       ocrLogger,
 			MonitoringEndpoint:           d.monitoringEndpointGen.GenMonitoringEndpoint(spec.ContractID),
 			OffchainConfigDigester:       keeperProvider.OffchainConfigDigester(),
 			OffchainKeyring:              kb,
 			OnchainKeyring:               kb,
-			ReportingPluginFactory: keeperreportingplugin.NewFactory(keeperreportingplugin.FactoryOptions{
-				Logger:          lggr.Named("OCR2Keeper"),
-				JobID:           jobSpec.ID,
-				ChainID:         chainID,
-				Cfg:             chain.Config(),
-				ORM:             orm,
-				EthClient:       chain.Client(),
-				HeadBroadcaster: chain.HeadBroadcaster(),
-				ContractAddress: spec.ContractID,
-				PipelineRunner:  d.pipelineRunner,
-				GasEstimator:    chain.TxManager().GetGasEstimator(),
-				PluginLimits: libocr2types.ReportingPluginLimits{
-					MaxQueryLength:       cfg.MaxQueryLength,
-					MaxObservationLength: cfg.MaxObservationLength,
-					MaxReportLength:      cfg.MaxReportLength,
-				},
-			}),
-		})
+			Registry:                     rgstry,
+			ReportEncoder:                ocr2keeperschain.NewEVMReportEncoder(),
+		}
+		del, err := ocr2keepers.NewDelegate(conf)
 		if err != nil {
-			return nil, errors.Wrap(err, "error calling NewOracle")
+			return nil, errors.Wrap(err, "could not create new keepers ocr2 delegate")
 		}
 
 		return []job.ServiceCtx{
 			runResultSaver,
 			keeperProvider,
-			registrySynchronizer,
```

Comment:
do we not need the registrySynchronizer anymore? We'll still need to listen to logs?

Reply:
In the future we will be running the synchronizer, but I think we should rebuild it when that time comes. There is a lot of stuff the synchronizer is doing that won't be necessary anymore such as database updates (unless we decide we do want to persist data). Even if we do decide to persist data, we can do so in a simpler way through the cache.

As for services associated with the plugin, we can decide in the future how we want to handle them. Maybe they should be part of the service context or maybe attached to the keeper delegate's Start/Close functions. OCR creates and destroys plugin instances at will so we have to be careful with the lifecycle of these services that outlive a single plugin instance.

Reply:
Agree we can skip writing to DB parts, but we still need some mechanism to listen to logs and update upkeeps

Reply:
Let's do that as a fast follower in the plugin.